mesonbuild/mparser.py

```diff
@@ -113,7 +116,7 @@ def __init__(self, code: str):
             ('whitespace', re.compile(r'[ \t]+')),
             ('multiline_fstring', re.compile(r"f'''(.|\n)*?'''", re.M)),
             ('fstring', re.compile(r"f'([^'\\]|(\\.))*'")),
-            ('id', re.compile('[_a-zA-Z][_0-9a-zA-Z]*')),
+            ('id', IDENT_RE),
```

Comment:
Wait a minute... There's really no reason for this to be an instance variable is there? It seems like we could get even more speedup by just making this a class variable? since that would avoid recompiling *any* regex and building the same list twice?

Reply:
True, but the cost of compiling these for the parser is not particularly high, only 15 ms for >200 calls in QEMU.

@bruchar1 is not wrong in that most of the benefit comes from rethinking the usage of regex and removing the matches completely; for example 30000 calls to match go away by removing unnecessary checks from variable assignments, and the cost for `set_variable` users is exactly the same as before..

Reply:
We create one instance of Parser, which internally contains one instance of Lexer, per `meson.build` file we parse. Moving it outside of `__init__` does feel cleaner indeed.